backend/src/manager/mod.rs

```diff
@@ -87,13 +87,17 @@ impl ManagerMap {
         manifest: Manifest,
         tor_keys: BTreeMap<InterfaceId, TorSecretKeyV3>,
     ) -> Result<(), Error> {
+        let keys = self.0.read().await.keys().cloned().collect::<Vec<_>>();
         let mut lock = self.0.write().await;
-        let id = (manifest.id.clone(), manifest.version.clone());
-        if let Some(man) = lock.remove(&id) {
-            if !man.thread.is_empty().await {
-                man.exit().await?;
+        for id in keys {
```

Comment:
Not sure that we want to leave the previous ones running and start another running service. Also, wondering why the keys are packageId and the Versions.